src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -90,19 +93,26 @@ void CGaussianKernel::set_width(float64_t w)
 
 SGMatrix<float64_t> CGaussianKernel::get_parameter_gradient(const TParameter* param, index_t index)
 {
+	using std::exp;
+
 	require(lhs, "Left hand side features must be set!");
-	require(rhs, "Rightt hand side features must be set!");
+	require(rhs, "Right hand side features must be set!");
 
 	if (!strcmp(param->m_name, "log_width"))
 	{
 		SGMatrix<float64_t> derivative=SGMatrix<float64_t>(num_lhs, num_rhs);
+		using EigenScalar = Eigen::Matrix<float64_t, 1, 1>;
+		Eigen::AutoDiffScalar<EigenScalar> eigen_log_width = m_log_width;
+		
 		for (int k=0; k<num_rhs; k++)
 		{
 #pragma omp parallel for
 			for (int j=0; j<num_lhs; j++)
 			{
-				float64_t element=distance(j, k);
-				derivative(j, k) = std::exp(-element) * element * 2.0;
+				eigen_log_width.derivatives() = EigenScalar::Unit(1,0);
+				auto el = CShiftInvariantKernel::distance(j, k);
+				Eigen::AutoDiffScalar<EigenScalar> kernel = exp(-el / (exp(eigen_log_width * 2.0) * 2.0));
```

Comment:
It kinda sucks that the code for the kernel needs to be in here as well as in the kernel itself dont you think?